application/F3DStarter.cxx

```diff
@@ -53,6 +67,8 @@ class F3DStarter::F3DInternals
   f3d::options DynamicOptions;
   f3d::options FileOptions;
   std::unique_ptr<f3d::engine> Engine;
+  std::vector<fs::path> FilesList;
+  long long CurrentFileIndex = -1;
```

Comment:
Obviously anything other than `std::optional<std::vector<fs::path>::size_type>` would be wrong and unacceptable ;)

what about `size_t` and have a separate variable to flag whether a file is loaded/active instead of using `-1`? (or keep using `-1`, it's good enough for `std::string::` with `npos`)

Reply:
just using an `int`  is fine I guess.